pkg/identity/cache/allocator.go

```diff
@@ -449,7 +453,7 @@ func (m *CachingIdentityAllocator) WatchRemoteIdentities(remoteName string, back
 		return nil, fmt.Errorf("error setting up remote allocator backend: %s", err)
 	}
 
-	remoteAlloc, err := allocator.NewAllocator(&key.GlobalIdentity{}, remoteAllocatorBackend, allocator.WithEvents(m.IdentityAllocator.GetEvents()))
+	remoteAlloc, err := allocator.NewAllocator(&key.GlobalIdentity{}, remoteAllocatorBackend, allocator.WithEvents(m.IdentityAllocator.GetEvents()), allocator.WithoutGC())
```

Comment:
What's the rationale of calling an allocator with GC? (Unfortunately, the existing code is super light on explanation for the GC part.)

On a related note, if an allocator is allocated with GC, then what does the stopping of the gc channel do in the `Delete()` - 

```
func (a *Allocator) Delete() {
	close(a.stopGC)
```

Reply:
if the GC is enabled.. the code starts this go routine:
```
if !a.disableGC {
		go func() {
			select {
			case <-a.initialListDone:
			case <-time.After(option.Config.AllocatorListTimeout):
				log.Fatalf("Timeout while waiting for initial allocator state")
			}
			a.startLocalKeySync()
		}()
	}
```
which in fact starts this go routine:
```
go func(a *Allocator) {
		kvTimer, kvTimerDone := inctimer.New()
		defer kvTimerDone()
		for {
			if err := a.syncLocalKeys(); err != nil {
				log.WithError(err).Warning("Unable to run local key sync routine")
			}

			select {
			case <-a.stopGC:
				log.Debug("Stopped master key sync routine")
				return
			case <-kvTimer.After(option.Config.KVstorePeriodicSync):
			}
		}
	}(a)
```
if I do not stop the `stopGC` channel I have no way of stopping the goroutine which is the problematic part.

So the GC (and these goroutines) only makes sense for the local allocator (as they sync the local keys to the backend) but only for that (as there are no local keys for the remote backend).